rules/__tests__/valid_expect_in_promise.test.js

```diff
@@ -0,0 +1,169 @@
+      errors: [
+        {
+          column: 24,
+          endColumn: 87,
+          message: expectedMsg,
+        },
+      ],
+    },
+    {
+      code:
+        'it("it1", function() { Promise.resolve().then(' +
+        'function() { /*fulfillment*/ expect(someThing).toEqual(true)}, ' +
+        'function() { /*rejection*/ expect(someThing).toEqual(true)})})',
+      errors: [
+        {
+          column: 24,
+          endColumn: 170,
+          message: expectedMsg,
+        },
+        {
+          column: 24,
+          endColumn: 170,
+          message: expectedMsg,
+        },
+      ],
+    },
+    {
+      code:
+        'it("it1", function() { Promise.resolve().then(' +
+        'function() { /*fulfillment*/}, ' +
+        'function() { /*rejection*/ expect(someThing).toEqual(true)})})',
+      errors: [
+        {
+          column: 24,
+          endColumn: 138,
+          message: expectedMsg,
+        },
+      ],
+    },
+  ],
+
+  valid: [
+    'it("it1", () => { return somePromise.then(() => {expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { return somePromise.catch(' +
+      'function() {expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { somePromise.then(' +
+      'function() {doSomeThingButNotExpect()})})',
+
+    'it("it1", function() { return getSomeThing().getPromise().then(' +
+      'function() {expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { return Promise.resolve().then(' +
+      'function() {expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { return Promise.resolve().then(' +
+      'function() { /*fulfillment*/ expect(someThing).toEqual(true)}, ' +
+      'function() { /*rejection*/ expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { return Promise.resolve().then(' +
+      'function() { /*fulfillment*/}, ' +
+      'function() { /*rejection*/ expect(someThing).toEqual(true)})})',
+
+    'it("it1", function() { return somePromise.then()})',
+
+    'it("it1", async () => { await Promise.resolve().then(' +
+      'function() {expect(someThing).toEqual(true)})})',
+
+    'it("it1", async () => ' +
+      '{ await somePromise.then(() => {expect(someThing).toEqual(true)})})',
+
+    'it("it1", async () => { await getSomeThing().getPromise().then(' +
+      'function() {expect(someThing).toEqual(true)})})',
+
+    "it('it1', () => {return somePromise." +
+      'then(() => {expect(someThing).toEqual(true);})' +
+      '.then(() => {expect(someThing).toEqual(true);})});',
+
+    "it('it1', () => {return somePromise." +
+      'then(() => {expect(someThing).toEqual(true);})' +
+      '.catch(() => {expect(someThing).toEqual(false);})});',
+    `
+     test('later return', () => {
+       const promise = something().then(value => {
+         expect(value).toBe('red');
+       });
+
+       return promise;
+     });
+    `,
+    `
+     it('shorthand arrow', () =>
+       something().then(value => {
+         expect(() => {
```

Comment:
this passes if I add `return` before `expect`

Reply:
I will work on this.

Reply:
There are two more similar cases,
1. Where expect is inside a block,
```
test('invalid return', () => {
       const promise = something().then(value => {
         { expect(value).toBe('red'); }
    });
 });
```
2. Where expect is inside another method,
```
test('invalid return', () => {
       const promise = something().then(value => {
         anotherMethod(); //this method contains the expect stament
    });
 });
```
I have already added this limitation(#>2) in the PR description. I think these are rare scenarios, the first one may not occur at all. As mentioned earlier in the PR description, I think they are `good to have` can be done in next enhancement to this rule.
If this scenarios are `must to have` , then I am happy to implement in this PR itself.